pkg/storage/unified/e2e/unified_scenario.go

```diff
@@ -0,0 +1,143 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/grafana/e2e"
+)
+
+type UnifiedScenario struct {
+	*e2e.Scenario
+
+	Grafana        *GrafanaService
+	UnifiedStorage *StorageService
+	Postgres       *PostgresService
+}
+
+func NewUnifiedScenario() (*UnifiedScenario, error) {
+	s, err := e2e.NewScenario(GetNetworkName())
+	if err != nil {
+		return nil, err
+	}
+
+	return &UnifiedScenario{
+		Scenario: s,
+	}, nil
+}
+
+func (s *UnifiedScenario) NewGrafanaService(name string, grpcEndpoint string) (*GrafanaService, error) {
+	flags := map[string]string{
+		"--config": fmt.Sprintf("/shared/%s.ini", name),
+	}
+	fts := []string{
+		"grafanaAPIServerWithExperimentalAPIs",
+		"kubernetesClientDashboardsFolders",
+		"grafanaAPIServerEnsureKubectlAccess",
+		"unifiedStorageSearch",
+		"unifiedStorageSearchUI",
+		"kubernetesCliDashboards",
+		"kubernetesDashboardsAPI",
+		"kubernetesFolders",
+		"unified_storage",
+	}
+	envVars := map[string]string{
+		"GF_PATHS_CONFIG":                   fmt.Sprintf("/shared/%s.ini", name),
+		"GF_FEATURE_TOGGLES_ENABLE":         strings.Join(fts, ","),
+		"GF_GRAFANA_APISERVER_ADDRESS":      grpcEndpoint,
+		"GF_GRAFANA_APISERVER_STORAGE_TYPE": "unified-grpc",
+		"GF_DATABASE_TYPE":                  "postgres",
+		"GF_DATABASE_HOST":                  "postgres:5432",
+		"GF_DATABASE_NAME":                  "grafana",
+		"GF_DATABASE_USER":                  "admin",
+		"GF_DATABASE_PASSWORD":              "admin",
+		"GF_DATABASE_SSL_MODE":              "disable",
+	}
+
+	err := s.loadCfg(name, grafanaINI)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Nit: we usually write those compacted if there is only an error returned. There are a few places throughout the code.
```suggestion
	if err := s.loadCfg(name, grafanaINI); err != nil {
		return nil, err
	}
```